src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -504,14 +505,17 @@ std::string postQueryContext
   //
   qcrsP->errorCode.fill(SccOk);
 
-  TIMED_MONGO(ciP->httpStatusCode = mongoQueryContext(qcrP,
-                                                      qcrsP,
-                                                      ciP->tenant,
-                                                      ciP->servicePathV,
-                                                      ciP->uriParam,
-                                                      ciP->uriParamOptions,
-                                                      countP,
-                                                      ciP->apiVersion));
+  if (limit != 0)
```

Comment:
You shouldn't do this check here. Note that you could need the calculation of `countP` done by mongoQueryContext() if the request is something like `GET /v2/entities?limit=0&options=count`. In fact, note step 4 in test/functionalTest/cases/1492_limit_zero/limit_zero_entities.test is not working correctly (you should get `Fiware-Total-Count: 2` instead of `Fiware-Total-Count: 0`).

Instead of that, I'd suggest you do deal with limit=0 situation internally in mongoQueryContext(). Maybe in entitiesQuery() or even in collectionRangedQuery().

Reply:
Fixed in https://github.com/telefonicaid/fiware-orion/pull/3994/commits/ff2b2251b1f0a0a1703c4e40b51d163f72bcede8